### ISSUE TITLE
MT Recall Mode: Label for relations

### ISSUE THREAD

**Issue by zifeishan** [HazyResearch/mindbender#29]
MT recall mode is useful for labeling mentions, but painful in labeling relations. As I mentioned in a separate email, currently to label for all relations in a document I need to:

1. hack the display of the mention_id in the bottom tags navbar
2. make a selection of mention 1, then copy the mention_id into a TSV file
3. make a selection of mention 2, then copy the mention_id into the same line of TSV file

It would be great to naturally support labeling recall in the interface. It would be simply nice to be able to:
* select the two mentions in the interface, then click a button, and the two mentioned are labeled as a relation. 

or alternatively:

* Select mention 1, click "add to relation"
* select mention 2,  click "add to relation"
* then click "finish labeling a relation" and they are recorded.

In the latter way one would be able to even label a relation in different pages.

As an idea to visualize relation, mentions in a relation can be marked up with a same number in the top-right corner.

@netj @chrismre

**Comment by netj** [comment]
Have you looked at the [genomics-recall-relation example](https://github.com/HazyResearch/mindbender/blob/master/examples/labeling/genomics-recall-relation/template.html) for marking up n-ary tags?

Adding an indicator next to the highlights sounds like a good idea.  I'll come up with a way to do that soon.

**Comment by zifeishan** [comment]
Wow this looks great. I will check it out. Thanks!

**Comment by zifeishan** [comment]
Works great!!

It seems that I still have some redundant error message, when the page is initially loading:

```
angular.js:11655 TypeError: Cannot read property 'address' of null
    at MultipleTag.enumerateValues (tags-parametric.coffee:141)
    at MultipleTag.enumerateValues (tags-parametric.coffee:1)
    at $parseFunctionCall (angular.js:12404)
    at $parseFieldAccess (angular.js:12338)
    at Object.expressionInputWatch (angular.js:12813)
    at Scope.$digest (angular.js:14300)
    at Scope.$apply (angular.js:14571)
    at done (angular.js:9698)
    at completeRequest (angular.js:9888)
    at XMLHttpRequest.requestLoaded (angular.js:9829)
```

I have one message per tag type. This does not affect usage though.

**Comment by netj** [comment]
Thanks for your input, I'll squash those errors as well soon.  So, does the relation labeling in the LATEST actually cover most of your previous hack?  I want to leave this issue open as a TODO for improving the visualization of relations anyway.

* [ ] Visualize each relation with an indicator, e.g., same number or symbol.

**Comment by zifeishan** [comment]
Yes, the LATEST release covers most of my hacks, except for rendering the split of documents.
